satpy/composites/cloud_products.py

```diff
@@ -51,17 +51,21 @@ def __call__(self, projectables, **info):
             raise ValueError("Expected 3 datasets, got %d" %
                              (len(projectables), ))
         data, palette, status = projectables
+        fill_value_color = palette.attrs.get("fill_value_color", [0, 0, 0])
         colormap, palette = self.build_colormap(palette, data.attrs)
-        channels = colormap.colorize(np.asanyarray(data))
-        not_nan = np.logical_and(self._get_mask_from_data(data), status != status.attrs['_FillValue'])
-        not_cloud_free = np.logical_or((status + 1) % 2, np.logical_not(not_nan))
-        chans = []
-        for channel in channels:
-            chan = self._create_masked_dataarray_like(channel, data, not_nan)
-            # Set cloud-free pixels as black
-            chans.append(chan.where(not_cloud_free, 0))
-
-        res = GenericCompositor.__call__(self, chans, **data.attrs)
+        mapped_channels = colormap.colorize(data.data)
+        valid = status != status.attrs['_FillValue']
+        # cloud-free pixels are marked invalid (fill_value in ctth_alti) but have status set to 1.
+        status_not_cloud_free = status % 2 == 0
```

Comment:
Isn't it onlu status=1 that indicates cloudfree?
Or is it a bit-flag?

Reply:
It is a bit-flag iiuc